src/libsyntax_ext/format.rs

```diff
@@ -65,10 +65,27 @@ struct Context<'a, 'b:'a> {
     /// Stays `true` if all formatting parameters are default (as in "{}{}").
     all_pieces_simple: bool,
 
-    name_positions: HashMap<String, usize>,
-
-    /// Updated as arguments are consumed
-    next_arg: usize,
+    /// Mapping between positional argument references and indices into the
+    /// final generated static argument array. We record the starting indices
+    /// corresponding to each positional argument, and number of references
+    /// consumed so far for each argument, to facilitate correct `Position`
+    /// mapping in `trans_piece`.
+    arg_index_map: Vec<Vec<usize>>,
+
+    /// Starting offset of count argument slots.
+    count_args_index_offset: usize,
+
+    /// Count argument slots and tracking data structures.
+    count_args: Vec<Position>,
+    /// Relative slot numbers for count arguments.
+    count_positions: HashMap<usize, usize>,
+    /// Number of count slots assigned.
+    count_positions_count: usize,
```

Comment:
I may be getting confused, but before there were no entries in this structure tracking `count_*` fields. How come these all cropped up now?